### PR TITLE
feat(securitycenter): Add Resource SCC Management API Org SHA Custom Module samples (Delete, Get Eff, List Eff, List Desc, Simulate)

### PR DESCRIPTION
**Pull request by vijaykanthm**
## Description

Fixes #b/347347517, b/347347514, b/347347515, b/347347453, b/347346959
Reference to b/347347517, b/347347514, b/347347515, b/347347453, b/347346959
This PR adds SCC Managament API Org SHA Custom Module Samples to Delete, Get Effective, List Effective, List Descendant & Simulate.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: securitycenter/management_api/delete_security_health_analytics_custom_module.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	// customModuleID := "your-module-id"
+	ctx := context.Background()
+	client, err := securitycentermanagement.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("securitycentermanagement.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	req := &securitycentermanagementpb.DeleteSecurityHealthAnalyticsCustomModuleRequest{
+		Name: fmt.Sprintf("%s/securityHealthAnalyticsCustomModules/%s", parent, customModuleID),
```
Comment (telpirion) [inline securitycenter/management_api/delete_security_health_analytics_custom_module.go:43]: nit(non-blocking): consider constructing the resource name outside of the request object. That way, customers can more clearly see and understand what the paths are in the resource name.

Here and below.

File: securitycenter/management_api/security_health_analytics_custom_module_test.go [hunk 2]
```diff
@@ -192,6 +192,33 @@ func cleanupExistingCustomModules(orgID string) error {
 	return nil
 }
 
+// TestDeleteCustomModule verifies the List functionality
+func TestDeleteCustomModule(t *testing.T) {
+	buf := new(bytes.Buffer)
```
Comment (telpirion) [inline securitycenter/management_api/security_health_analytics_custom_module_test.go:197]: nit(non-blocking): consider just declaring the buffer like so:

```go
var buf bytes.Buffer
```

You don't need to use the `new()` keyword.

File: securitycenter/management_api/security_health_analytics_custom_module_test.go [hunk 3]
```diff
@@ -213,6 +240,34 @@ func TestCreateCustomModule(t *testing.T) {
 	}
 }
 
+// TestListDescendantCustomModule verifies the List Descendant functionality
+func TestListDescendantCustomModule(t *testing.T) {
+	buf := new(bytes.Buffer)
```
Comment (telpirion) [inline securitycenter/management_api/security_health_analytics_custom_module_test.go:245]: Same as previous.

File: securitycenter/management_api/security_health_analytics_custom_module_test.go [hunk 4]
```diff
@@ -242,6 +297,55 @@ func TestGetCustomModule(t *testing.T) {
 	}
 }
 
+// TestSimulateCustomModule verifies the Create functionality
+func TestSimulateCustomModule(t *testing.T) {
+	buf := new(bytes.Buffer)
```
Comment (telpirion) [inline securitycenter/management_api/security_health_analytics_custom_module_test.go:302]: Same as previous.

File: securitycenter/management_api/security_health_analytics_custom_module_test.go [hunk 5]
```diff
@@ -269,6 +373,35 @@ func TestUpdateCustomModule(t *testing.T) {
 	}
 }
 
+// TestGetEffectiveCustomModule verifies the Get Effective functionality
+func TestGetEffectiveCustomModule(t *testing.T) {
+	buf := new(bytes.Buffer)
```
Comment (telpirion) [inline securitycenter/management_api/security_health_analytics_custom_module_test.go:378]: Same as previous.

File: securitycenter/management_api/simulate_security_health_analytics_custom_module.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+						Description: "The name of the instance",
+						Expression:  "resource.name",
+						Location:    "global",
+						Title:       "Instance Name",
+					},
+				},
+			},
+		},
+		Description: "Sample custom module for testing purpose. Please do not delete.", //Replace with the desired description.
+		Predicate: &expr.Expr{
```
Comment (telpirion) [inline securitycenter/management_api/simulate_security_health_analytics_custom_module.go:61]: issue: move comment to above the field initialization. The comment is difficult to read when it's on the same line as what it's commenting.

**Review by telpirion — APPROVED** [review]
Mostly minor nits, one small formatting issue. Thank you!

File: securitycenter/management_api/delete_security_health_analytics_custom_module.go (hunk 1 above)
Comment (vijaykanthm) [reply securitycenter/management_api/delete_security_health_analytics_custom_module.go]: Addressed.

File: securitycenter/management_api/security_health_analytics_custom_module_test.go (hunk 2 above)
Comment (vijaykanthm) [reply securitycenter/management_api/security_health_analytics_custom_module_test.go]: Addressed.

File: securitycenter/management_api/security_health_analytics_custom_module_test.go (hunk 3 above)
Comment (vijaykanthm) [reply securitycenter/management_api/security_health_analytics_custom_module_test.go]: Addressed.

File: securitycenter/management_api/security_health_analytics_custom_module_test.go (hunk 4 above)
Comment (vijaykanthm) [reply securitycenter/management_api/security_health_analytics_custom_module_test.go]: Addressed.

File: securitycenter/management_api/security_health_analytics_custom_module_test.go (hunk 5 above)
Comment (vijaykanthm) [reply securitycenter/management_api/security_health_analytics_custom_module_test.go]: Addressed.

File: securitycenter/management_api/simulate_security_health_analytics_custom_module.go (hunk 6 above)
Comment (vijaykanthm) [reply securitycenter/management_api/simulate_security_health_analytics_custom_module.go:62]: Addressed.

**Pull request merged**